source/vstd/invariant.rs

```diff
@@ -233,6 +233,44 @@ declare_invariant_impl!(LocalInvariant);
 #[cfg_attr(verus_keep_ghost, verifier::proof)]
 pub struct InvariantBlockGuard;
 
+// In the "Logical Paradoxes" section of the Iris 4.1 Reference
+// (`https://plv.mpi-sws.org/iris/appendix-4.1.pdf`), they show that
+// opening invariants carries the risk of unsoundness. One solution to
+// this, described in the paper "Later Credits: Resourceful Reasoning
+// for the Later Modality" by Spies et al. (available at
+// `https://plv.mpi-sws.org/later-credits/paper-later-credits.pdf`) is
+// to use "later credits". That is, require the expenditure of a later
+// credit, only obtainable in exec mode, when opening an invariant. So
+// we require the relinquishment of a tracked
+// `OpenInvariantCredit` to open an invariant, and we provide an
+// exec-mode function `create_open_invariant_credit` to obtain one.
+
+verus! {
+
+#[doc(hidden)]
+#[cfg_attr(verus_keep_ghost, verifier::proof)]
+#[verifier::external_body]
+pub struct OpenInvariantCredit {}
+
+// It's intentional that `create_open_invariant_credit` uses `exec` mode. This prevents
+// creation of an infinite number of credits to open invariants infinitely often.
+#[cfg(verus_keep_ghost)]
+#[verifier::external_body]
+#[inline(always)]
+pub fn create_open_invariant_credit() -> Tracked<OpenInvariantCredit>
```

Comment:
You may want a `rustc_diagnostic_item` on this too, to match with it in `modes.rs` (see the other comment).